pkg/rancher-desktop/config/settings.ts

```diff
@@ -427,3 +430,114 @@ function updateSettings(settings: Settings) {
+ * @returns A settings profile stripped of all fields not found in the schema.
+ */
+export function readDeploymentProfile() {
+  let profileData;
+  let profile;
+
+  switch (os.platform()) {
+  case 'win32':
+    try {
+      profile = readRegistryDeploymentProfile(); // TODO: call with path to system profile
+    } catch {
+      try {
+        profile = readRegistryDeploymentProfile(); // TODO: call with path to user profile
+      } catch {}
+    }
+    break;
+  case 'linux':
+    try {
+      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_JSON), 'utf8');
+    } catch {
+      try {
+        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_JSON), 'utf8');
+      } catch {}
+    }
+    if (typeof profileData !== 'undefined') {
+      profile = JSON.parse(profileData);
+    }
+    break;
+  case 'darwin':
+    try {
+      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_PLIST), 'utf8');
+    } catch {
+      try {
+        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_PLIST), 'utf8');
+      } catch {}
+    }
+    if (typeof profileData !== 'undefined') {
+      profile = plist.parse(profileData);
+    }
+    break;
+  }
+
+  return validateDeploymentProfile(profile, defaultSettings);
+}
+
+function readRegistryDeploymentProfile() {
+  /* Not implemented */
+}
+
+function validateDeploymentProfile(profile: any, schema: any) {
+  if (typeof profile === 'undefined') {
+    return undefined;
+  }
+
+  Object.keys(profile).forEach((key) => {
+    if (key in schema) {
+      if (typeof profile[key] === typeof schema[key]) {
+        if (typeof profile[key] === 'object') {
+          validateDeploymentProfile(profile[key], schema[key]);
+        }
+      } else {
+        console.log(`Deployment Profile ignoring '${ key }'. Wrong type.`);
+        delete profile[key];
+      }
+    } else {
+      console.log(`Deployment Profile ignoring '${ key }'. Not in schema.`);
+      delete profile[key];
+    }
+  });
+
+  return profile;
+}
+
+/**
+ * Merge profile into config and return result.
+ * @param {*} config The config settings file.
+ * @param {*} profile The deployment profile.
+ * @param {*} lockedOnly Only overwrite settings for locked profile values.
+ * @return {*} The merged configuration file.
+ */
+export function mergeDeploymentProfile(config: any, profile: any, lockedOnly: boolean): any {
+  if (typeof config === 'undefined' || typeof profile === 'undefined') {
+    return (profile || config);
+  }
+
+  if (lockedOnly === true) {
+    config = _.mergeWith(config, profile, (configObj, profileObj) => {
+      for (const key of Object.keys(profileObj)) {
+        if (typeof profileObj[key] === 'object') {
+          for (const [k, v] of Object.entries(profileObj[key])) {
+            if (k === 'locked' && v === true) {
+              console.log(`Deployment Profile overriding locked '${ key }'`);
```

Comment:
This should be `console.debug` because it's normal behaviour.

Reply:
Agreed.  changed.